### PR TITLE
Refresh page on consent change

### PR DESCRIPTION
**Pull request by emma-imber**
## What does this change?
Adds a module to refresh the page when consent state changes, and an E2E test to wait for a page reload when consent state changes.

Also deletes the module to remove consented ads on consent withdrawal, as we no longer need this if we're refreshing the page immediately.

## Why?
This means our data will correctly reflect the page consent status, and that users will immediately experience the appropriate ad experience on consent change.

### CODE REVIEW

**Review by Jakeii — APPROVED** [review]
Nice! Finally over the line!

File: src/init/shared/reload-page-on-consent-change.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		if (initialConsentState === undefined) {
+			initialConsentState = consent;
+		}
+		if (initialConsentState.canTarget !== consent.canTarget) {
+			window.location.reload();
+		}
+	}, true);
+};
+
+const reloadPageOnConsentChange = once(() =>
```
Comment (arelra) [inline src/init/shared/reload-page-on-consent-change.ts:22]: Not sure why we need to once() as I thought we would only call this in the boot script?

Comment (emma-imber) [reply src/init/shared/reload-page-on-consent-change.ts]: Good spot - I think this is from the original PR back in frontend. I've removed it and seems to work fine without!

**Review by arelra — APPROVED** [review]
🥳 

ps @zekehuntergreen 😉

**Pull request merged**